### DISCUSSION TITLE
Cannot access /Volumes on mac from within vm

### DISCUSSION THREAD

**Discussion by EdwardMoyse** [lima-vm/lima#1660] [Q&A]
My usecase is that the default filesystem for macs is case insensitive, but I need case sensitive. So I created a new volume: /Volumes/Lima

And then in my yaml I have:
```
mounts:
  - location: "~"
  - location: /Volumes/Lima
    writable: true
```

But on starting the vm I see:
```
ERRO[0037] [1 error occurred:
	* failed to prepare reverse sshfs for "/Users/emoyse/Lima" on "/Users/emoyse/Lima": failed to mkdir "/Users/emoyse/Lima" (remote): "mkdir: cannot create directory ‘/Users/emoyse/Lima’: File exists\n": exit status 1

] 
```

I also tried using
```
  - location: ~/Lima
```
where `~/Lima` was a symbolic link to `/Volumes/Lima`

In both cases I could not access it from within the vm. Is this not possible?

**Comment by afbjorklund** [comment]
You can of course use case-sensitive Mac disk image (instead of Linux), but I don't see how /Volumes conflict with /Users.

Do you have the rest of the debug log, from why it fails when you try to map `/Volumes/Lima` (without the symlink) ?

**Reply by EdwardMoyse** [reply]
Sorry, I think the issue was I was trying to map `~/Lima` and `/Volumes/Lima` at the same time...

Out of interest, which debug log would I look in (the message above was just what appeared when I did `limactl start`)?

**Reply by afbjorklund** [reply]
I just meant `limactl --debug` output

**Reply by EdwardMoyse** [reply]
Ah - I see! Thanks, that will be useful for next time.

**Comment by jandubois** [comment]
The error message seems to come from trying to mount `~/Lima`, which will not work. You shouldn't use overlapping mounts (your already map `~`), and you need to mount the target of the symlink, not the symlink itself.

I have no idea why mounting `/Volumes/Lima` wouldn't work; we would need to see the actual error from that config.

**Reply by EdwardMoyse** [reply]
Thank you for your reply! Probably I was mixing up log output from two separate attempts (my apologies if so!)

Trying again with:
```
mounts:
  - location: "~"
  - location: /tmp/lima
    writable: true
  - location: /Volumes/Lima
    writable: true
```
and deleting ~/Lima, and it all seems to work! Thanks again for your help (and sorry for the noise).

One thing I don't quite understand though, inside the VM `df -h` gives me:
```
Filesystem      Size  Used Avail Use% Mounted on
/dev/vda3       200G   69G  132G  35% /
:/Users/emoyse  746G  334G  412G  45% /Users/emoyse
:/tmp/lima      746G  334G  412G  45% /tmp/lima
:/Volumes/Lima  746G  334G  412G  45% /Volumes/Lima
```

whilst from the host, I get:
```
Filesystem       Size   Used  Avail Capacity iused      ifree %iused  Mounted on
/dev/disk4s1s1  745Gi  8.4Gi  412Gi     3%  355382 4292246204    0%   /
...
/dev/disk4s7    745Gi  1.4Gi  412Gi     1%   75608 4314927520    0%   /Volumes/Lima
```
This looks more correct - I only have 1.4 GiB in `/Volumes/Lima` so I'm not sure why it's misreported inside the VM?

**Reply by afbjorklund** [reply]
> I'm not sure why it's misreported inside the VM?

Could be a quirk with sshfs and the sftp server, maybe it is reporting the statfs of the mount point instead of the file system ?

Are you using the builtin SFTP server, or the external `openssh-sftp-server`. Can it be reproduced with SSH (sftp) perhaps ?
